src/routes/instances/+page.server.ts

```diff
@@ -0,0 +1,52 @@
+import type { PageServerLoad } from "./$types";
+
+export const load: PageServerLoad = async () => {
+	const invidious = await fetch("http://localhost:5173/").catch((err) => {
+		return err;
+	});
+	const librarian = await fetch("https://lbry.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const libreddit = await fetch("https://libreddit.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const nitter = await fetch("https://nitter.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const element = await fetch("https://chat.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const piped = await fetch("https://piped.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const searxng = await fetch("https://search.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const gitea = await fetch("https://git.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const portainer = await fetch("https://portainer.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const mailcow = await fetch("https://mail.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+	const plausible = await fetch("https://analytics.projectsegfau.lt/").catch((err) => {
+		return err;
+	});
+
+	return {
+		updated: Math.floor(Date.now() / 1000),
+		invidious: invidious.status || invidious.code,
+		librarian: librarian.status || librarian.code,
+		libreddit: libreddit.status || libreddit.code,
+		nitter: nitter.status || nitter.code,
+		element: element.status || element.code,
+		piped: piped.status || piped.code,
+		searxng: searxng.status || searxng.code,
+		gitea: gitea.status || gitea.code,
+		portainer: portainer.status || portainer.code,
+		mailcow: mailcow.status || mailcow.code,
+		plausible: plausible.status || plausible.code
+	};
+};
```

Comment:
Honestly i prefer having something that could not literally fetch every website status before returning, and no soleil levant won't be able to load this fast, You should also consider the situation where a website is not even respondiing, say a connection timeout.

Reply:
The problem is that when i tested it there was 0 indication whatsoever that it was loading, i thought the button or link broke, so i clicked multiple times, which itself causes more unnecessary requests. Worst case scenario add a loading button ? (And noJS users don't even get the status page report). Another approach is to load the page and then ping every service and THEN output the result (again noJS users won't benefit from it but it's the price to pay for fast page loading.)